pydantic/_internal/_decorators.py

```diff
@@ -252,11 +253,13 @@ class Decorator(Generic[DecoratorInfoType], Representation):
 
     def __init__(
         self,
+        cls_ref: str,
```

Comment:
@adriangb As long as you are cool with adding this attribute, I am good with this PR.